### PR TITLE
Adding etc_hosts virtual table to windows

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 0955788f0f533c58e945ed15dc048c19c3597511 (Job results: [69](https://jenkins.osquery.io/job/osqueryPullRequestAudit/69/)) failed the code audit.

File: osquery/tables/networking/etc_hosts.cpp [hunk 1]
```diff
@@ -24,6 +24,11 @@
 namespace osquery {
 namespace tables {
 
+#ifndef WIN32
+std::string kEtcHosts = "/etc/hosts";
+#else
+std::string kEtcHosts = "C:\\Windows\\System32\\drivers\\etc\\hosts";
```
Comment (mtmcgrew) [inline osquery/tables/networking/etc_hosts.cpp]: dynamically grab the c:\windows dir if it's changed. GetWindowsDirectory and PathCombine can help

**Comment by osqueryer** [comment]
:-1: The commit 0955788f0f533c58e945ed15dc048c19c3597511 (Job results: [3086](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3086/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:no_good: The commit f048969009dff649384478eb4e02de9e7b971d79 (Job results: [74](https://jenkins.osquery.io/job/osqueryPullRequestAudit/74/)) failed the code audit.

**Comment by osqueryer** [comment]
:-1: The commit f048969009dff649384478eb4e02de9e7b971d79 (Job results: [109](https://jenkins.osquery.io/job/osqueryPullRequestBuild/109/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit f048969009dff649384478eb4e02de9e7b971d79 (Job results: [3091](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3091/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 98d92302031d67eef6ae595237583f5563f72028 (Job results: [3118](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3118/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 98d92302031d67eef6ae595237583f5563f72028 (Job results: [133](https://jenkins.osquery.io/job/osqueryPullRequestBuild/133/)) failed one or more tests (**Windows**).

File: osquery/tables/networking/etc_hosts.cpp [hunk 2]
```diff
@@ -21,9 +22,17 @@
...
+#define fs boost::filesystem
+
 namespace osquery {
 namespace tables {
 
+#ifndef WIN32
+fs::path kEtcHosts = "/etc/hosts";
+#else
+fs::path kEtcHosts =
+    (getSystemRoot() / "system32\\drivers\\etc\\hosts").string();
```
Comment (theopolis) [inline osquery/tables/networking/etc_hosts.cpp]: You can drop the `.string()` here.

**Comment by osqueryer** [comment]
:no_good: The commit 3222930685b2819d70e20f2d946d3b574bcc2781 (Job results: [106](https://jenkins.osquery.io/job/osqueryPullRequestAudit/106/)) failed the code audit.

**Comment by osqueryer** [comment]
:-1: The commit 3222930685b2819d70e20f2d946d3b574bcc2781 (Job results: [3123](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3123/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit e3e73f818548ae621a0624259f2599ba3c913b0d (Job results: [3125](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3125/)) failed one or more tests (**OS X/Linux**).

**Comment by PoppySeedPlehzr** [comment]
```
[+] brew package rocksdb upgrading to new version: 4.6.1
Error: No available formula with the name "rocksdb" 
make: *** [deps] Error 1
Build step 'Execute shell' marked build as failure
Set build name.
New build name is 'bc288de74f2a0ef1251074e26cee1f465c58fdc4'
Variable with name 'BUILD_DISPLAY_NAME' already exists, current value: 'bc288de74f2a0ef1251074e26cee1f465c58fdc4', new value: 'bc288de74f2a0ef1251074e26cee1f465c58fdc4'
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
Finished: FAILURE
```

**Pull request merged**